### PR TITLE
Refactor/runtime tests with new test utilities

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**
no work on the tests attribute.test.ts and attribute-e2e.test.ts because Milena is doing them

(Presumably) a race condition for the syncUntil-function causes tests to occasionally fail.

### CODE REVIEW

**Review by jkoenig134 — CHANGES_REQUESTED** [review]
Only one small comment. Rest LGTM.

File: packages/runtime/test/lib/testUtils.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export async function syncUntilHasMessageWithRequest(transportServices: Transpor
 
 export async function syncUntilHasMessageWithResponse(transportServices: TransportServices, requestId: string | CoreId): Promise<MessageDTO> {
     const filterResponseMessagesByRequestId = (syncResult: SyncEverythingResponse) => {
-        return syncResult.messages.filter((m) => m.content["@type"] === "ResponseWrapper" && m.content.requestId === requestId.toString());
+        return syncResult.messages.filter((m) => (m.content["@type"] === "Response" || m.content["@type"] === "ResponseWrapper") && m.content.requestId === requestId.toString());
```
[the remark this answers is not in the document]
Comment (Magnus-Kuhn) [reply packages/runtime/test/lib/testUtils.ts:90]: the runtime test "sender: sync Message with Response" in requests.test.ts sends a Response

[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/runtime/test/lib/testUtils.ts:90]: you should switch to ResponseWrapper there.

File: packages/runtime/test/consumption/iqlQuery.test.ts [hunk 2]
```diff
@@ -320,20 +313,13 @@ describe("IQL Query", () => {
...
-    });
-
-    test("sender: complete the outgoing Request with Response from Message", async () => {
+    test("sender: sync Message with Response and complete the outgoing Request with Response from Message", async () => {
         let triggeredEvent: OutgoingRequestStatusChangedEvent | undefined;
         sEventBus.subscribeOnce(OutgoingRequestStatusChangedEvent, (event) => {
             triggeredEvent = event;
         });
 
+        sResponseMessage = await syncUntilHasMessageWithResponse(sTransportServices, sLocalRequest.id);
```
Comment (slavistan) [inline packages/runtime/test/consumption/iqlQuery.test.ts:322]: This `sReponseMessage` should be a local variable (as well as a lot of others which are defined in the header of the "IQL Query" test suite), which is part of a bigger problem: the tests share state, which is not something we want, as that makes it impossible to run and debug a single test in isolation.

Should be rectified either in this PR or a separate one.

Comment (slavistan) [reply packages/runtime/test/consumption/iqlQuery.test.ts:322]: I'm still uncertain about if the tests run in parallel or not. If some of these tests are in fact run in parallel sharing state between them would explain any occasional test failures.

Comment (jkoenig134) [reply packages/runtime/test/consumption/iqlQuery.test.ts:322]: @slavistan tests does never run in parallel, that's only the case for test suites.

Comment (slavistan) [reply packages/runtime/test/consumption/iqlQuery.test.ts:322]: I created a separate backlog item for these coupled tests.

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**